src/lib.rs

```diff
@@ -79,6 +79,18 @@ where
 
         self.frontmatter
     }
+    /// Iterate over the FrontmatterExtractor iterator and stop after it has finished
+    /// searching for frontmatter. Returns the Iterator so you can do additional parsing
+    /// on the rest of the document. The optional first H1 title header and frontmatter
+    /// code block have been consumed
+    pub fn iterate_and_return(mut self) -> FrontmatterExtractor<'a, T> {
+        while let Some(_) = self.next() {
```

Comment:
If the document doesn't contain frontmatter, I believe this will cause an Event to be dropped and never returned from the iterator.

Reply:
Ah, you mean if there's an h1 header?

Reply:
Yes, actually after digging into this further, I don't see a better way to implement this without causing this crate to add extra allocations. This is because the extractor only removes the code block while returning the other events such as the header or the document's first event.

Since this approach is going to require adding allocations, I think adding a new API that extracts the frontmatter in one step and retains the intermediate events in a Vec would be a good approach. The `DocumentAttributeParserState` can have a new variant which could contain the vec's drain iterator to pop events off of in the `Iterator::next` implementation.